scripts/check_docs

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#
+# Check that Rust documentation is generated without any warnings.
+
+readonly SCRIPTS_DIR="$(dirname "$(readlink -f "$0")")"
+# shellcheck source=scripts/common
+source "$SCRIPTS_DIR/common"
+
+# `cargo doc` produces warnings for incorrect paths. These warnings cannot be promoted to errors, so we use grep to detect them.
+if cargo doc --document-private-items --no-deps 2>&1 | grep --ignore-case --quiet --regexp='^warning'; then
+  echo "Warnings found when generating the docs."
+  exit 1
+fi
+
+# Check for any deadlinks in the generated docs. 
+if ! cargo deadlinks --dir target/doc; then
```

Comment:
Is this `--dir` needed? (just asking because it was not there before)

Reply:
I think so. The one in `build_gh_pages` did not do anything on my machine.

Reply:
Are you sure the issue isn't that you have some pre-generated stuff under target/doc, and that's what's causing it to fail? AFAICT those failures you mentioned are related to some tracing stuff that we don't actually depend on. But I may be wrong.

Reply:
I thought about that, but it happened on the [cloudbuild](https://pantheon.corp.google.com/cloud-build/builds/3b65e797-8b79-47df-8ee0-4326647d9724?project=oak-ci) too (this is a build for a version that used the `--check-http` option). So, I don't think the errors are due to some pre-generated docs.

Reply:
Without the `--dir` option, `cargo deadlinks` only checks the files in `abitest_common`.

Reply:
Oh, why that particular one? Is that just the first one in lexicographic order, or does it have something special?

Reply:
The documentation says that the default directory is `target/doc/<package>`. I guess when the `package` is not specified it picks up the first one in some order. The `--verbose` option outputs the files that are being checked.